### ISSUE TITLE
Proposal: Have a check for undocumented Public functions

### ISSUE THREAD

**Issue by lswith** [rrrene/credo#452]
A simple lint rule to say whether there are undocumented public functions in the codebase or not would be very helpful. Inch_ex seemed to suggest it did this but it doesn't return any error codes and seems to be unmaintained.

**Comment by bradleygolden** [comment]
@rrrene, I opened up a PR for this mostly mirroring the [module doc](https://github.com/rrrene/credo/blob/master/lib/credo/check/readability/module_doc.ex) check.

Some basic assumptions that I made are:

1. Sometimes people might want to ignore a public function, though I can't think of a case where you wouldn't use `@doc false` instead. Any case you can think of would be helpful as I'm fairly new to elixir.
2. Empty string docs will always throw a warning even if it's ignored because having an empty string doc is just strange.

**Comment by bradleygolden** [comment]
Any thoughts on what I should do for the case where someone has provided documentation metadata but no doc attribute for the description?

Throws a warning `Use @doc false if a function will not be documented.`:

```elixir
@doc since: "1.3.0"
def public_func do
  "something"
end
```

Should I still throw a warning that the user need to provide an @doc description?

Doesn't throw a warning:

```elixir
@doc "Some description"
@doc since: "1.3.0"
def public_func do
  "something"
end
```

**Comment by rrrene** [comment]
@bradleygolden The check you are creating has the same "problems" the original `@moduledoc` check has: It does not capture module attributes created via meta programming, since Credo does not compile the project it analyses.

Any rules you might come up with have probably already been covered by [`inch_ex`](https://github.com/rrrene/inch_ex) which has been updated to `2.0` since this issue was filed.

**Comment by bradleygolden** [comment]
Makes sense... I wasn't aware of inch_ex prior to this. Should this issue be closed out in lieu of using inch ex for documentation related checks?
